File: app/src/main/java/de/westnordost/streetcomplete/quests/DateUtil.java
package de.westnordost.streetcomplete.quests;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public class DateUtil {
	public static SimpleDateFormat basicISO8601(){
		return new SimpleDateFormat("yyyy-MM-dd", Locale.US);

	}
	public static String getOffsetDateStringFromDate(final int offsetInDays, final Date date){
		Calendar modifiedCalendar = Calendar.getInstance();
		modifiedCalendar.setTime(date);
		modifiedCalendar.add(Calendar.DAY_OF_MONTH, offsetInDays);
		return basicISO8601().format(modifiedCalendar.getTime());
	}

	public static String getOffsetDateString(int offsetInDays){
		return getOffsetDateStringFromDate(offsetInDays, Calendar.getInstance().getTime());
	}

	public static String getCurrentDateString() {
		return getOffsetDateString(0);
	}
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/DateUtil.kt
package de.westnordost.streetcomplete.quests

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Locale

object DateUtil {

    fun getCurrentDateString(): String = getOffsetDateString(0)

    fun basicISO8601() = SimpleDateFormat("yyyy-MM-dd", Locale.US)

    /**
     * negative offsetInDays returns dates from past, positive from future
     */
    fun getOffsetDateStringFromDate(offsetInDays: Int, date: Date): String {
        val modifiedCalendar = Calendar.getInstance()
        modifiedCalendar.time = date
        modifiedCalendar.add(Calendar.DAY_OF_MONTH, offsetInDays)
        return basicISO8601().format(modifiedCalendar.time)
    }

    /**
     * negative offsetInDays returns dates from past, positive from future
     */
    fun getOffsetDateString(offsetInDays: Int) =
        getOffsetDateStringFromDate(offsetInDays, Calendar.getInstance().time)
}
